app/models/registration_payment.rb

```diff
@@ -10,14 +10,25 @@ class RegistrationPayment < ApplicationRecord
   has_many :refunding_registration_payments, class_name: 'RegistrationPayment', inverse_of: :refunded_registration_payment, foreign_key: :refunded_registration_payment_id, dependent: :destroy
 
   delegate :auto_accept_preference_live?, to: :registration
+  before_save :set_paid_at, if: :becoming_completed?
   after_create :auto_accept_hook, if: :auto_accept_preference_live?
-  after_create :auto_close_hook, unless: :refunded_registration_payment_id?
+  after_save :auto_close_hook
+
+  scope :completed, -> { where(is_completed: true) }
 
   monetize :amount_lowest_denomination,
            as: "amount",
            allow_nil: true,
            with_model_currency: :currency_code
 
+  private def becoming_completed?
+    is_completed && (will_save_change_to_is_completed? || new_record?)
+  end
+
+  private def set_paid_at
+    self.paid_at = Time.now.utc unless self.paid_at.present?
```

Comment:
```suggestion
    self.paid_at = current_time_from_proper_timezone unless self.paid_at.present?
```

This is a Rails helper that is also used when calling `touch` on a model

Reply:
Also, the `unless` part can probably move directly into the `before_save` part. Something along the lines of `unless: :paid_at?`